src/test/java/RecipeTest.java

```diff
@@ -203,7 +198,7 @@ public void toStringTest() {
     expected += "\nSteps:\n";
     expected += "\tA step\n";
 
-    Recipe testRecipe = new Recipe("NAME", "DESC", Arrays.asList(new Step("A step")));
+    Recipe testRecipe = new Recipe("NAME", "DESC", TAGS, INGREDIENTS, Arrays.asList(new Step("A step")));
     Assert.assertEquals(expected, testRecipe.toString());
   }
 }
```

Comment:
I think it would be nice to add a test that ensures that the exception you put in place (invalid step handler) is actually thrown (https://www.baeldung.com/junit-assert-exception)

Reply:
All the tests that are marked @Test(expected = IndexOutOfBoundsException.class) do this, the invalid step handler method just throws an IndexOutOfBoundsException after logging the index.

Reply:
o my bad i totally skipped over that

Reply:
ty for clarification!